camel/toolkits/hybrid_browser_toolkit/ws_wrapper.py

```
@@ -0,0 +1,487 @@
+        """Ensure WebSocket connection is alive."""
+        if not self.websocket:
+            raise RuntimeError("WebSocket not connected")
+
+        # Check if connection is still alive
+        try:
+            # Send a ping to check connection
+            await self.websocket.ping()
+        except Exception as e:
+            logger.warning(f"WebSocket ping failed: {e}")
+            self.websocket = None
+            raise RuntimeError("WebSocket connection lost")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)